tools/run_onnx.cc

```diff
@@ -341,14 +341,20 @@ void RunMain(const std::vector<std::string>& argv) {
     chainerx::ContextScope ctx_scope(ctx);
     chainerx::NoBackpropModeScope no_backprop;
     const std::string device_spec = args.get<std::string>("device");
+    bool enable_memory_monitoring = false;
     if (!device_spec.empty()) {
         chainerx::Device* device = &chainerx::GetDefaultContext().GetDevice(device_spec);
         chainerx::SetDefaultDevice(device);
         if (IsCudaDevice(device)) {
             g_use_cuda = true;
             g_meminfo_enabled = true;
+            if (g_compiler_log) {
+                InitializeMemoryMonitoring(device);
```

Comment:
It sounds reasonable. I will try to move this to `ChxVM::Prepare`.

Reply:
Well, I changed my mind and I want to keep current implementation (at least as for initialization part).
When we want to benchmark memory usage from python code, we need to run `InitializeMemoryMonitoring` at the beginning of python code (otherwise some chx arrays won't be tracked.) In python code, `ChxVM::Prepare` will be called at the intermediate point of entire process. The change seems to make it difficult to benchmark entire memory consumption from python code.

Reply:
Yeah, I agree `initialize_memory_monitoring` Python interface is fine. I just wanted to move more code from here to runtime.